[13.0][MIG] stock_available_mrp

ps-tubtim opened the pull request:
This PR is waiting for https://github.com/OCA/stock-logistics-warehouse/pull/802, you can always use this for readability.
- [x] [13.0][MIG] stock_available

sbidoul committed:
[MOV] move addons out of __unported__ (they remain not installable)

lmignon committed:
[FIX] Browse mrp.bom as current user
sudo is not required since mrp.bom are readable to groups with access to the qty_x fields on a product. Moreover using sudo to retrive the bom will ignore the company_id defined on the bom

Commit:
[MIGR] stock_available_mrp: migrate to v8

Compute potential quantities for both product templates and variants. To keep the code simple, only the biggest potential of any single variant is accounted for in the template's potential.
Take all levels of phantom BoM into account, respects validity dates etc. thanks to the use of the standard method _bom_explode, as suggested by @gdgellatly in https://github.com/OCA/stock-logistics-warehouse/pull/5#issuecomment-66902191
Improve tests, rewritten in python.
Adhere to new file/manifest/README conventions.
Simplify copyright headers

Commit:
[FIX] fix multi-company test

Record rules used to not be checked on stock quants, but now they are since Odoo's commit 2fd14db57433d08ab368a7e18d18ae8d49dafab1 (https://github.com/odoo/odoo/commit/2fd14db).
In our test we changed the company of the products and BoMs but we neglected that the stock was not attached to the right company, and that made the test fail.
To fix that, make the test inventory for the right company.
Since there is a little inconsistency in the demo data with a negative quantity of an unrelated product, use the `partial` filter for the inventories instead of the `none` filter, so that no wrong inventory lines are added automatically.

oca-transbot committed:
OCA Transbot updated translations from Transifex

Commit:
stock_available / stock_available_mrp : add possibility to choose which field use to compute potential

Commit:
stock_available_mrp: fix BOM qty (multi-units, efficiency)

Commit:
[FIX+IMP] stock_available_mrp: fix recursive potential_qty in list view + small improvements

Commit:
V9 migration modifications.

    * mrp_bom.name has been deleted.
    * mrp_bom_line.type moved to mrp_bom.type.
    * Fix missing group_mrp_user issue.
    * Change versions

oca-transbot committed:
OCA Transbot updated translations from Transifex

Commit:
[10.0][MIG] port stock_available_mrp to 10.0

rousseldenis committed:
[IMP] stock_available_mrp: readability + change compute bom_id + use recordset + Condition never statisfied

Commit:
[ADD] include new improvments

Commit:
[FIX] stock_available_mrp: remove duplicate button + fix calculation of immediately usable qty

Commit:
[11.0][MIG] port stock_available_mrp to 11.0

florian-dacosta committed:
Migrate stock_available_mrp to v12

florian-dacosta committed:
Make stock_available_mrp compatible with other modules like stock_available_immediately

pedrobaeza committed:
[FIX] stock_available_mrp: Fix RST formatting + generate README

oca-travis committed:
[UPD] Update stock_available_mrp.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-warehouse-12.0/stock-logistics-warehouse-12.0-stock_available_mrp
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-12-0/stock-logistics-warehouse-12-0-stock_available_mrp/

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-warehouse-12.0/stock-logistics-warehouse-12.0-stock_available_mrp
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-12-0/stock-logistics-warehouse-12-0-stock_available_mrp/

liweijie0812 committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (7 of 7 strings)

Translation: stock-logistics-warehouse-12.0/stock-logistics-warehouse-12.0-stock_available_mrp
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-12-0/stock-logistics-warehouse-12-0-stock_available_mrp/zh_CN/

m-abdalrahman commented:
hello @ps-tubtim I can't enter the products screen after installing the module
![Screenshot from 2020-06-15 14-18-38](https://user-images.githubusercontent.com/19738046/84656705-66134600-af13-11ea-8cbd-a19fd7bacd12.png)

log:
`odoo.http: ('product.product(1,).bom_id', None)
`

after add store=True to bom_id field to fix the previous issue, when I Create new product potential quantities values not calculated, but if I uninstall stock_available and install stock_available_mrp  potential quantities values calculated and appear

rousseldenis commented:
@ps-tubtim You can rebase as #802 is merged

i-vyshnevska commented:
@ps-tubtim can you pls rebase and check travis build for errors

leemannd commented:
Hello @ps-tubtim a fix is included into https://github.com/OCA/stock-logistics-warehouse/pull/1070

leemannd commented on stock_available_mrp/models/product_product.py:
```diff
@@ -0,0 +1,139 @@
+                or (not b.product_id and b.product_tmpl_id == product.product_tmpl_id)
+            )
+            if product_boms:
+                product.bom_id = first(product_boms)
```
```suggestion
                product.bom_id = first(product_boms)
            else:
                product.bom_id = False
```

ps-tubtim replied:
Thank you for your review. Edited.

leemannd reviewed (commented):
Missing returb

leemannd reviewed (changes requested):
Missing return in compute function. This will lead to the impossibility to see products

chienandalu committed:
[FIX] stock_available_mrp: upstream compatibility

After this change https://github.com/odoo/odoo/commit/3d34d58388ab362bf8e69cc2e75adbdfda3331b2
it isn't possible to have an storable kit product anymore. The main
issue is that we have to drop some uses cases that this module
considered for this kind of situations.

OCA-git-bot committed:
stock_available_mrp 12.0.1.0.1

chienandalu commented:
We just merged this changes due to Odoo upstream changes that intruduced some incompatibilities. https://github.com/OCA/stock-logistics-warehouse/pull/1099/files Please include it in your migration

ps-tubtim committed:
[IMP] stock_available_mrp: black, isort, prettier

ps-tubtim committed:
[MIG] stock_available_mrp: Migration to 13.0

leemannd commented:
This can be closed. The module has been migrated into another PR
